src/mocks/timeseriesDataPointsList.ts

```diff
@@ -1,9 +1,95 @@
 import {
   DatapointsGetAggregateDatapoint,
   DatapointsGetDoubleDatapoint,
+  GetAggregateDatapoint,
 } from '@cognite/sdk';
+import { MockChartDataConfig } from '.';
 
-export function randomData(
+export const defaultConf: MockChartDataConfig = {
+  id: 0,
+  min: 0,
+  max: 150,
+  continousDeviation: 3,
+  peakDeviation: 30,
+};
+
+export function getRandomdata(
+  id: number,
+  start: number,
+  end: number,
+  n: number,
+  granularity?: number,
+  config?: MockChartDataConfig[]
+) {
+  const conf = config ? getConfig(config, id) : defaultConf;
+  const arr = randomData(start, end, n, conf, granularity);
+  return insertPeakPoints(arr, conf);
+}
+
+function getConfig(config: MockChartDataConfig[], id: number) {
+  const configuration = config.find(conf => conf.id === id);
+  return configuration ? configuration : defaultConf;
+}
+
+function insertPeakPoints(
+  resultArray: DatapointsGetAggregateDatapoint,
+  config: MockChartDataConfig
+) {
+  const { positivePeakPoints, negativePeakPoints, peakDeviation } = config;
+  const positiveDPs = positivePeakPoints ? positivePeakPoints : 0;
+  const negativeDPs = negativePeakPoints ? negativePeakPoints : 0;
+  const { datapoints: datapointList } = resultArray;
+  const randomIndexesList = generateRandomIndexesList(
+    datapointList.length - 1,
+    positiveDPs + negativeDPs
+  );
+
+  let positiveDPsVal = positiveDPs;
+  while (positiveDPsVal !== 0) {
+    const index = randomIndexesList.pop();
+    const dp = datapointList[index ? index : 0];
+    addPeakDeviation(dp, peakDeviation);
+    positiveDPsVal -= 1;
+  }
+
+  let negativeDPsVal = negativeDPs;
+  while (negativeDPsVal !== 0) {
+    const index = randomIndexesList.pop();
+    const dp = datapointList[index ? index : 0];
+    substractPeakDeviation(dp, peakDeviation);
+    negativeDPsVal -= 1;
+  }
+
+  return resultArray;
+}
+
+function addPeakDeviation(dp: GetAggregateDatapoint, peakDeviation: number) {
```

Comment:
This function is to add peak points to the generated data. So I thought it is unnecessary to make a deep copy of this. Do we really need to make this function immutable?

Reply:
It's not so functional approach to mutate props in general. It would be better to stick to the gearbox codestyle.

`GetAggregateDatapoint` is a pretty shallow type, so it isn't a problem to copy it and return new object.